### PR TITLE
feat: add show hide password icon

### LINKED ISSUE #2467: [feature request] Show Password Icon
**Is your feature request related to a problem? Please describe.**
When I make a mistake, especially at the end, it irritates me that I either forgot to save the password for quick access or that I was in a rush. I believe being able to view the password you type would be helpful.

**Describe the solution you'd like**
The password would initially be hidden, i.e. with stars. We have in the right corner of the input a button ( icon ) that show/hide the password

**Additional context**
I quickly implemented this to demonstrate how it will seem.

Group 106

### PR DESCRIPTION
**Pull request by BaltacMihai**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

I made a new component that contains a Field and an icon that changes its type. I hope I have included it everywhere where we have a field of the password type.

The icons at the moment are just for testing, I'm waiting for the new icons to replace what we have.

## Git Issues

Closes #2467

### CODE REVIEW

**Comment by davehakkens** [comment]
Noticed that the error text is inbetween the icon and input field.
Would make more sense if its after te icon


afbeelding

File: src/common/Form/PasswordField.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { useState } from 'react'
+import { Field } from 'react-final-form'
+import { Icon } from 'oa-components'
+import { Box } from 'theme-ui'
+
+export const PasswordField = ({ name, component, ...rest }) => {
+  const [isPasswordVisible, setIsPasswordVisible] = useState(false)
```
Comment (AlfonsoGhislieri) [inline src/common/Form/PasswordField.tsx:7]: **Praise**: nice job pulling this out into a separate component!

**Review by AlfonsoGhislieri — APPROVED** [review]
Thanks @BaltacMihai this looks good, great work!

File: src/common/Form/PasswordField.tsx (hunk 1 above)
Comment (BaltacMihai) [reply src/common/Form/PasswordField.tsx:7]: Thank you!

**Pull request merged**